components/script/script_module.rs

```diff
@@ -1199,80 +1202,86 @@ fn fetch_module_descendants(
     module_tree: &ModuleTree,
     destination: Destination,
     mut visited: HashSet<ServoUrl>,
-) -> Result<Vec<Rc<Promise>>, ()> {
+) -> Result<Vec<Rc<Promise>>, ModuleError> {
     println!("Start to load dependencies of {}", module_tree.url.clone());
 
     let global = owner.global();
 
     module_tree.set_status(ModuleStatus::FetchingDescendants);
 
-    let requested_urls = module_tree.resolve_requested_modules(&global, &mut visited);
-
-    if let Ok(requested_urls) = requested_urls {
-        module_tree.set_descendant_urls(requested_urls.clone());
-
-        println!("{:?}", requested_urls);
+    module_tree
+        .resolve_requested_modules(&global, &mut visited)
+        .map(|requested_urls| {
```

Comment:
this is getting way too indented and really hard to read, can we have something like

```
let requested_urls = if let Some(requested) = module_tree.resolve(..)) {
   reqested
} else {
  return None
}
```

in general there are a lot of uses of `.map` and stuff leading to tons of indentation making it hard to read, it would be nice if you could de-indent things wherever possible

Reply:
Let me think about how to fix this `map` 🤔